From 455087b99190834b3361b38a58d59032c90acc1e Mon Sep 17 00:00:00 2001
From: Jonathan Kim <jonathan.kim@amd.com>
Date: Mon, 12 Aug 2019 11:47:28 -0400
Subject: [PATCH 0822/1453] drm/amdkfd: updated read to set errno to eagain

User space requries -1 and errno set to EAGAIN on kfifo_to_user
failure or 0 bytes copy.

v2: ported missing commit to amd-staging-dkms-5.2

Change-Id: I34985ff641b21568e12bf5534bd39d1d5d744364
Signed-off-by: Jonathan Kim <Jonathan.Kim@amd.com>
Signed-off-by: Felix Kuehling <Felix.Kuehling@amd.com>
Reviewed-by: Jonathan Kim <Jonathan.Kim@amd.com>
---
 drivers/gpu/drm/amd/amdkfd/kfd_debug_events.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdkfd/kfd_debug_events.c b/drivers/gpu/drm/amd/amdkfd/kfd_debug_events.c
index f549f1d8f6ef..df98f30d66d7 100644
--- a/drivers/gpu/drm/amd/amdkfd/kfd_debug_events.c
+++ b/drivers/gpu/drm/amd/amdkfd/kfd_debug_events.c
@@ -70,9 +70,10 @@ static ssize_t kfd_dbg_ev_read(struct file *filep, char __user *user,
 
 	ret = kfifo_to_user(&dpd->fifo, user, size, &copied);
 
-	if (ret) {
-		pr_debug("KFD DEBUG EVENT: Failed to read poll fd (%i)\n", ret);
-		return ret;
+	if (ret || !copied) {
+		pr_debug("KFD DEBUG EVENT: Failed to read poll fd (%i) (%i)\n",
+								ret, copied);
+		return ret ? ret : -EAGAIN;
 	}
 
 	return copied;
-- 
2.17.1

